task-generator/remote/main.go

```diff
@@ -96,14 +99,34 @@ func convertToSsh(task *tektonapi.Task) {
 			syncVolumes[i.Name] = true
 		}
 	}
+	// The images produced in multi-platform builds need to have unique tags in order
+	// to prevent them from getting garbage collected before generating the image index.
+	// We can simplify this process, preventing the need for users to manually specify
+	// the image by auto-appending the architecture from the PLATFORM parameter. For
+	// example, this will append -arm64 if PLATFORM is linux/arm64. Since we cannot
+	// modify the parameter itself, this replacement needs to happen in any task step
+	// where the IMAGE parameter is used.
```

Comment:
so for existing builds, will we end up with $image-s390x-s390x (since they must have manually added the suffix already?) I suppose that is ugly but harmless?

Reply:
No, I made the comment clearer. If the suffix does not exist on the `IMAGE` then we will add it. If the suffix does exist, we will not duplicate it.